MovieGallery/Pages/Search.cshtml.cs

```diff
@@ -44,28 +46,36 @@ public void OnGet(string SearchbyYear)
             {
                 Query = SearchbyYear;
                 long inputYear = Convert.ToInt64(Query);
-
-                var result = MoviesJson.Items.Where(x => x.Year == inputYear).ToList();
-                MoviesResult = new Movies()
+                if (inputYear > 1000 && inputYear <= Convert.ToInt64(currentYear))
                 {
-                    Items = result
-                };
+                    var result = MoviesJson.Items.Where(x => x.Year == inputYear).ToList();
+                    MoviesResult = new Movies()
+                    {
+                        Items = result
+                    };
 
-                var resultShow = ShowsJson.Items.Where(x => x.Year == inputYear).ToList();
-                ShowsResult = new Shows()
-                {
-                    Items = resultShow
-                };
+                    var resultShow = ShowsJson.Items.Where(x => x.Year == inputYear).ToList();
+                    ShowsResult = new Shows()
+                    {
+                        Items = resultShow
+                    };
 
-                if (MoviesResult.Items.Count > 0 || ShowsResult.Items.Count > 0)
-                {
-                    SearchCompleted = true;
+                    if (MoviesResult.Items.Count > 0 || ShowsResult.Items.Count > 0)
+                    {
+                        SearchCompleted = true;
+                        CorrectYear = true;
+                    }
+                    else
+                    {
+                        SearchCompleted = false;
+                        CorrectYear = true;
+                        MoviesResult = new Movies();
+                        ShowsResult = new Shows();
+                    }
                 }
                 else
                 {
-                    SearchCompleted = false;
-                    MoviesResult = new Movies();
-                    ShowsResult = new Shows();
+                    CorrectYear = false;
```

Comment:
Making the correctyear as false and this validation will be checked in html file and accordingly the error message will be displayed.